internal/clients/embed.go

```diff
@@ -0,0 +1,54 @@
+package clients
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+var AwsSupported = make(map[string]interface{})
+var ErrArchitectureNotSupported = errors.New("architecture is not supported")
+
+const x86_64 = "x86_64"
+const arm64 = "arm"
+const i386 = "i386"
+
+type supportedInstanceTypes struct {
+	Aws struct {
+		Ec2 []string `yaml:"ec2"`
```

Comment:
I still don't get this nesting btw :)

Reply:
Yeah this is weird, I suggest to always have some root map, and then just use `ec2`. Something like:


```yaml
allowed_types:
  ec2:
    - x1.super-tiny
    - x9.huge-xxxxl
  azure:
    - z13.windows.laptop
```

Reply:
Are you saying to use `aws` instead of `ec2`? Sure, if you prefer this term just use it!